release

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env -S bash -e
+
+cd $(dirname $0)
+
+main() {
+  local version="$1"
+  local preview_txid="$2"
+  local preprod_txid="$3"
+
+  [[ -z "$version" ]]      && usage "Missing version number"
+  [[ -z "$preview_txid" ]] && usage "Missing preview network hydra scripts publication transaction id"
+  [[ -z "$preprod_txid" ]] && usage "Missing preprod network hydra scripts publication transaction id"
+
+  check_can_release "$version"
+
+  prepare_release "$version" "$preview_txid" "$preprod_txid"
+
+  publish_release # fake for now
+}
+
+usage() {
+  local message=$1
+
+  cat >&2 <<EOF
+$message
+
+  $0 <version> <preview_txid> <preprod_txid>
+Publishes a new release of hydra.
+<version> must respect [Semantic Versioning](http://semver.org/)
+<preview_txid> is the id of the transaction used to publish the hydra scripts on the preview network (see smoke tests)
+<preprod_txid> is the id of the transaction used to publish the hydra scripts on the preprod network (see smoke tests)
+EOF
+
+  exit 1
+}
+
+check_can_release() {
+  local version="$1"
+
+  check_no_uncommited_changes  
+  check_version_is_valid $version
+  check_changelog_is_up_to_date $version
+
+  true #avoid error on last instruction of function (see bash -e)
+}
+
+prepare_release() {
+  local version="$1"
+  local preview_txid="$2"
+  local preprod_txid="$3"
+
+  update_cabal_version "$version"
+  update_api_version   "$version"
+  update_preview_txid "$version" "$preview_txid"
+  update_preprod_txid "$version" "$preprod_txid"
```

Comment:
I don't se much use of having this aliased functions. I'd suggest inlining the `update_txid preview "$version" "$preview_txid"` directly.

Reply:
At this level of code I appreciate the fact that we don't have to know how the functions work to understand that we will update the preview taxied. The fact that it's almost the same code is an implementation detail.
I could probably refactor to have only one single function and keep it as obvious to read here but I'd like to see how we integrate mainnet first before touching this as, I suspect, we gone have subtil nuances to integrate when managing mainnet and I'm afraid refactoring now would make it harder then. Oh well... let's say I want to see what happen with mainnet.